sql/log.cc

```diff
@@ -626,8 +626,9 @@ bool write_bin_log_start_alter(THD *thd, bool& partial_alter,
     start_alter_info *info= thd->rgi_slave->sa_info;
     bool is_shutdown= false;
 
-    info->sa_seq_no= start_alter_id;
-    info->domain_id= thd->variables.gtid_domain_id;
+    info->gtid.domain_id= thd->variables.gtid_domain_id;
+    info->gtid.server_id= thd->variables.server_id;
+    info->gtid.seq_no= start_alter_id;
```

Comment:
`start_alter_id` originates from `get_start_alter_id()`, which gives `thd->variables.gtid_seq_no` in a (configured) replica and `0` otherwise.
And there’re code that branches on this `0` or proceeds otherwise.

Perhaps the replica-checking logic should be split away from `start_alter_id`, so `THD::variables` can hold a GTID multi-member like `start_alter_info` in this PR?
Implementation’s out of this ticket’s scope, of course.